doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,241 @@
+arma::Row<size_t> predictions;
+gb.Classify(testDataset, predictions);                  // Step 3: classify points.
+
+// Print some information about the test predictions.
+std::cout << arma::accu(predictions == 2) << " test points classified as class "
+    << "2." << std::endl;
+```
+<p style="text-align: center; font-size: 85%"><a href="#simple-examples">More examples...</a></p>
+
+#### Quick links:
+
+ * [Constructors](#constructors): create `GradientBoosting` objects.
+ * [`Train()`](#training): train model.
+ * [`Classify()`](#classification): classify with a trained model.
+ * [Other functionality](#other-functionality) for loading, saving, and
+   inspecting.
+ * [Examples](#simple-examples) of simple usage and links to detailed example
+   projects.
+
+#### See also:
+
+ * [`AdaBoost`](adaboost.md)
+ * [`DecisionTree`](decision_tree.md)
+ * [`DecisionTreeRegressor`](decision_tree_regressor.md)
+ * [Random forests](random_forest.md)
+ * [mlpack classifiers](../../index.md#classification-algorithms)
+ * [Gradient Boosting on Wikipedia](https://en.wikipedia.org/wiki/Gradient_boosting)
+
+### Constructors
+
+ * `gb = GradBoosting()`
+   - Initialize model without training.
+   - You will need to call [`Train()`](#training) later to train the model before
+     calling [`Classify()`](#classification).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical and/or categorical data using default arguments for the
+     weak learners ([`DecisionTreeRegressor`s](decision_tree_regressor.md)).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical and/or categorical data, using the given hyperparameters.
+
+---
+
+#### Constructor Parameters:
+
+| **name** | **type** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| `data` | [`arma::mat`](../matrices.md) | [Column-major](../matrices.md#representing-data-in-mlpack) training matrix. | _(N/A)_ |
+| `datasetInfo` | [`data::DatasetInfo`](https://github.com/mlpack/mlpack/blob/master/doc/user/load_save.md#loading-categorical-data) | Dataset information, specifying type information for each dimension. | _(N/A)_ |
+| `labels` | [`arma::Row<size_t>`](../matrices.md) | Training labels, [between `0` and `numClasses - 1`](../load_save.md#normalizing-labels) (inclusive).  Should have length `data.n_cols`.  | _(N/A)_ |
+| `numClasses` | `size_t` | Number of classes in the dataset. | _(N/A)_ |
+| `numWeakLearners` | `size_t` | Number of weak learner models to be used. | 5 |
+| `minimumLeafSize` | `size_t` | Minimum leaf size for weak learner decision tree. | `10` |
+| `minimumGainSplit` | `double` | Minimum gain split for weak learner decision tree | `1e-7` |
+| `maximumDepth` | `size_t` | Maximum tree depth for weak learner decision tree | `2` |
+
```

Comment:
It would be useful to add some advice here on how to set these hyperparameters.  Up to you whether you want to take these suggestions, but I think they are right:

```suggestion
 * Smaller values of `numWeakLearners` and `maximumDepth` will result in smaller `GradientBoosting` models, but the model may underfit the data.
 * Smaller values of `minimumLeafSize` and `minimumGainSplit` will improve the performance of the `GradientBoosting` model, but at the cost of training time and model size.
```

(Also maybe `minLeafSize` and `minGainSplit` and `maxDepth` would be good shorter names that still get the same idea across.)